### ISSUE TITLE
CUDA MLA (DeepSeek-V2/V3) RoPE ignores YaRN scaling — long-context correctness gap

### ISSUE THREAD

**Issue by jamesburton** [jamesburton/dotLLM#430]
## Summary

CUDA's MLA (DeepSeek-V2/V3) attention builds its decoupled-RoPE cos/sin tables with the plain, non-YaRN routine, while the CPU MLA path uses the YaRN-aware routine. This is the same class of CPU/CUDA divergence that issue #366 fixed for dense RoPE, but on a separate code path that #366 explicitly did not touch.

## Evidence

- `src/DotLLM.Cuda/CudaTransformerModel.cs:772` builds the MLA decoupled-rope tables with:
  ```
  DotLLM.Cpu.Kernels.RoPE.PrecomputeFrequencyTable(maxSeq, mla.QkRopeHeadDim, mla.RopeTheta, cosArr, sinArr);
  ```
  — the **plain** (non-YaRN) routine, unconditionally.

- `src/DotLLM.Models/Architectures/TransformerModel.cs:338-358` builds the CPU MLA tables, gated on `MlaConfig.RopeScalingFactor > 1`, calling `RoPE.PrecomputeFrequencyTableYarn` at `:353` in that case.

So any DeepSeek-V2/V3 checkpoint that ships `rope_scaling.factor > 1` (long-context YaRN) will get CPU and CUDA logits that diverge, because CUDA silently keeps using the plain frequency ramp.

## Severity — contrast with #366's dense-RoPE bug

#366 fixed a dense-RoPE bug where the missing `mscale` multiplier caused a ~34.7% Q/K divergence **at position 0** (a short-context, immediately-visible error) for gpt-oss and any other dense-YaRN model (SmolLM3-128k, Llama 3.1+ extended context).

MLA's YaRN mscale is different: it's a **ratio** `yarn_get_mscale(factor, mscale) / yarn_get_mscale(factor, mscale_all_dim)`, which is 1.0 whenever `mscale == mscale_all_dim` — the common case (e.g. DeepSeek-V2-Lite). So this MLA gap is predominantly a **frequency-ramp / long-context** correctness gap (positions beyond `original_max_position_embeddings` diverge), not a position-0 error. It is real but harder to observe on short prompts than the bug #366 fixed.

## Fix shape

Unlike #366's dense-RoPE fix (which required threading `inv_freq`/`mscale` through 3 CUDA kernels and PTX rebuilds), MLA never touches those kernels — it consumes precomputed cos/sin tables built once per model (`_mlaRopeCosF32`/`_mlaRopeSinF32`, sized `[maxSeq, qkRopeHeadDim/2]`, rebuilt on growth at `CudaTransformerModel.cs:759-779`). The fix should be confined to `CudaTransformerModel.EnsureMlaState`: switch the table-build call to `RoPE.PrecomputeFrequencyTableYarn` when `MlaConfig.RopeScalingFactor > 1`, mirroring `TransformerModel.cs:338-358` exactly. No kernel or PTX work, no allocation-lifetime change.

## References

- Sibling fix: #366 (CUDA SWA + dense-YaRN RoPE for gpt-oss) — this issue's gap was found and documented during that investigation but deliberately left unfixed as out of scope.
- Working notes: `.docs/366-rope-scaling-findings.md` on branch `issue/366-cuda-swa-alternating` (git-ignored) has the full derivation, including the mscale-ratio formula and `RoPE.cs:181-184` reference for when `mscale == mscale_all_dim`.
- Comments at `src/DotLLM.Cuda/CudaWeights.cs` (near `RopeYarnInvFreqDevice` / `UploadDenseYarnInvFreq`) already record that MLA is explicitly excluded from #366's fix and why.
